packages/flutter_tools/lib/src/macos/xcdevice.dart

```diff
@@ -112,8 +112,15 @@ class XCDevice {
         throwOnError: true,
       );
       if (result.exitCode == 0) {
-        final List<dynamic> listResults = json.decode(result.stdout) as List<dynamic>;
-        _cachedListResults = listResults;
+        List<dynamic> listResults = <dynamic>[];
+        final String listOutput = result.stdout;
+        try {
+          listResults = json.decode(listOutput) as List<dynamic>;
+          _cachedListResults = listResults;
+        } on FormatException {
+          // xcdevice logs errors and crashes to stdout.
+          _logger.printError('xcdevice returned non-JSON response: $listOutput');
```

Comment:
should we return null here, as we do below if `result.exitCode != 0`?

Reply:
Good idea.

Reply:
Done.